src/psyclone/tests/domain/lfric/transformations/raise_psyir_2_lfric_alg_trans_test.py

```diff
@@ -208,6 +208,32 @@ def test_codeblock_invalid(monkeypatch, fortran_reader):
             "Structure-Constructor, but found 'NoneType'." in str(info.value))
 
 
+def test_arg_declaration_error(fortran_reader):
+    '''Test that the validate method raises an exception if the argument
+    to the invoke is a builtin whose name has already been used as a
+    subroutine name.
+
+    '''
+    code = (
+        "subroutine setval_c()\n"
+        "  use builtins\n"
+        "  use constants_mod, only: r_def\n"
+        "  use field_mod, only : field_type\n"
+        "  type(field_type) :: field\n"
+        "  real(kind=r_def) :: value\n"
+        "  call invoke(setval_c(field, value))\n"
+        "end subroutine setval_c\n")
+    psyir = fortran_reader.psyir_from_source(code)
+    invoke_trans = RaisePSyIR2LFRicAlgTrans()
+    with pytest.raises(TransformationError) as info:
+        invoke_trans.validate(psyir.children[0][0])
+    assert ("The arguments to this invoke call are expected to be kernel "
+            "calls which are represented in generic PSyIR as CodeBlocks "
+            "or ArrayReferences, but 'setval_c(field, value)' is of type "
+            "'Call'. Has 'setval_c' also been used as a routine name in "
```

Comment:
Can we do better? i.e. presumably it ends up as a `Call` because we already have a `RoutineSymbol` named 'setval_c'. It might be better if we could say this as well as saying which SymbolTable we found it in?